contracts/governance/Governor.sol

```diff
@@ -760,68 +762,24 @@ abstract contract Governor is Context, ERC165, EIP712, Nonces, IGovernor, IERC72
         address proposer,
         string memory description
     ) internal view virtual returns (bool) {
-        uint256 len = bytes(description).length;
+        uint256 length = bytes(description).length;
 
         // Length is too short to contain a valid proposer suffix
-        if (len < 52) {
+        if (length < 52) {
             return true;
         }
 
-        // Extract what would be the `#proposer=0x` marker beginning the suffix
-        bytes12 marker;
-        assembly ("memory-safe") {
-            // - Start of the string contents in memory = description + 32
-            // - First character of the marker = len - 52
-            //   - Length of "#proposer=0x0000000000000000000000000000000000000000" = 52
-            // - We read the memory word starting at the first character of the marker:
-            //   - (description + 32) + (len - 52) = description + (len - 20)
-            // - Note: Solidity will ignore anything past the first 12 bytes
-            marker := mload(add(description, sub(len, 20)))
-        }
+        // Extract what would be the `#proposer=` marker beginning the suffix
+        bytes10 marker = bytes10(Bytes.unsafeReadBytesOffset(bytes(description), length - 52));
 
         // If the marker is not found, there is no proposer suffix to check
-        if (marker != bytes12("#proposer=0x")) {
+        if (marker != bytes10("#proposer=")) {
```

Comment:
doesn't look beautiful but we can save the casting cost with the hardcoded value, and add a comment for visibility
```suggestion
        if (marker != 0x2370726f706f7365723d00000000000000000000000000000000000000000000) {
```

Reply:
the good thing that it allows to show that if the proposer is zero it returns true as well

Reply:
AFAIK there is no casting cost here. Have you been able to observe any difference?

Reply:
![Capture d’écran du 2024-10-11 21-38-59](https://github.com/user-attachments/assets/47965f6d-69a1-4b26-9941-f4c30efca998)

I tried these two, they produce the exact same bytecode

Reply:
okk makes sense. thanks for checking!

Reply:
![Capture d’écran du 2024-10-11 21-40-18](https://github.com/user-attachments/assets/c40363f5-59a9-47b0-a15b-67860705a7ad)

I tried these two, they produce the exact same bytecode